AnnService/src/Core/Common/DistanceUtils.cpp

```diff
@@ -416,14 +416,34 @@ float DistanceUtils::ComputeL2Distance_SSE(const std::uint8_t* pX, const std::ui
     }
     float diff = DIFF128[0] + DIFF128[1] + DIFF128[2] + DIFF128[3];
 
+    float c1;
+    uint8_t a, b;
     while (pX < pEnd4) {
-        float c1 = ((float)(*pX++) - (float)(*pY++)); diff += c1 * c1;
-        c1 = ((float)(*pX++) - (float)(*pY++)); diff += c1 * c1;
-        c1 = ((float)(*pX++) - (float)(*pY++)); diff += c1 * c1;
-        c1 = ((float)(*pX++) - (float)(*pY++)); diff += c1 * c1;
+	a = (*pX); b = (*pY);
+	c1 = a + 0.0f - b;
```

Comment:
Could you share a bit more detail what's the difference in behavior here (floating point quirks?) to me th elogic looks the same

Reply:
It is the AMD cpu compiling issue. When using the original two lines it will cause illegal instruction segmentation fault.